### ISSUE TITLE
test_perf_profiler: test_trampoline_works_with_forks() failed on Address Sanitizer CI

### ISSUE THREAD

**Issue by vstinner** [python/cpython#109580]
# Bug report

```
test_trampoline_works_with_forks (test.test_perf_profiler.TestPerfTrampoline.test_trampoline_works_with_forks) ... FAIL

FAIL: test_trampoline_works_with_forks (test.test_perf_profiler.TestPerfTrampoline.test_trampoline_works_with_forks)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cpython/cpython/Lib/test/test_perf_profiler.py", line 123, in test_trampoline_works_with_forks
    self.assertEqual(process.returncode, 0)
AssertionError: 139 != 0
```

It looks like a SIGSEGV crash (128 + SIGSEGV = 139 on Linux).

build: https://github.com/python/cpython/actions/runs/6237694188/job/16931875912?pr=108965

Passed when re-run:

```
0:09:17 load avg: 4.36 Re-running 1 failed tests in verbose mode in subprocesses
0:09:17 load avg: 4.36 Run 1 test in parallel using 1 worker process (timeout: 20 min, worker timeout: 25 min)
0:09:17 load avg: 4.36 [1/1] test_perf_profiler passed
Re-running test_perf_profiler in verbose mode (matching: test_trampoline_works_with_forks)
test_trampoline_works_with_forks (test.test_perf_profiler.TestPerfTrampoline.test_trampoline_works_with_forks) ... ok

----------------------------------------------------------------------
Ran 1 test in 0.083s

OK
```

cc @pablogsal 

### Linked PRs
* gh-109584
* gh-109585

**Comment by vstinner** [comment]
I cannot reproduce the issue locally on Fedora 38.

One option is to skip the test if Python is built with ASAN, as already done on TestPerfProfiler test case: skip the whole file.

**Comment by pablogsal** [comment]
> I cannot reproduce the issue locally on Fedora 38.
> 
> 
> 
> One option is to skip the test if Python is built with ASAN, as already done on TestPerfProfiler test case: skip the whole file.

I'm fine skipping the file if is built with ASAN. In any case this file should only run in buildbots where perf is available.

**Comment by vstinner** [comment]
@pablogsal:
> I'm fine skipping the file if is built with ASAN.

Ok, I created PR #109584 for that.

I don't have the bandwidth to investigate why Python built with ASAN does crash on this Ubuntu Address Sanitizer job, but it doesn't when I run the test locally on my Fedora 38.
